crawler/src/main/java/com/javaacademy/crawler/App.java

```diff
@@ -13,57 +11,49 @@
 
 import java.io.IOException;
 import java.io.InputStream;
-import java.util.Properties;
-import java.util.Set;
+import java.util.*;
 
 import static com.javaacademy.crawler.common.logger.AppLogger.DEFAULT_LEVEL;
 import static com.javaacademy.crawler.common.logger.AppLogger.logger;
-import static com.javaacademy.crawler.common.util.CrawlerUtils.sleepFor;
 
 /**
  * @author devas
  * @author mprtcz
  * @since 24.08.17
  */
 public class App {
+    private static Map<String, Long> crawlersSendBooksStats = new HashMap<>();
+    private static String serverIpAddress;
 
     public static void main(String[] args) {
         AppLogger.initializeLogger();
-        String serverIpAddress = loadIpAddress();
+        serverIpAddress = loadIpAddress();
         if (serverIpAddress.equals("")) return;
-
-        Set<BookModel> bonitoBooks = new BonitoScrapper().scrape();
-        BookSender bonitoBookSender = new BookSender(bonitoBooks);
-        bonitoBookSender.sendBooksTo(serverIpAddress, "Bonito");
-
-        Set<BookModel> gandalfBooks = new GandalfScrapper().scrape();
-        BookSender gandalfBooksSender = new BookSender(gandalfBooks);
-        gandalfBooksSender.sendBooksTo(serverIpAddress, "Gandalf");
-
-        Set<BookModel> matrasBooks = new MatrasScrapper().scrape();
-        BookSender matrasBooksSender = new BookSender(matrasBooks);
-        matrasBooksSender.sendBooksTo(serverIpAddress, "Matras");
-
-        Set<BookModel> czytamBooks = new CzytamScrapper().scrape();
-        BookSender czytamBooksSender = new BookSender(czytamBooks);
-        czytamBooksSender.sendBooksTo(serverIpAddress, "Czytam");
-
-        Set<Book> googleBooks = runGoogleScrapper();
-        BookSender googleBookSender = new BookSender(googleBooks, new GoogleBookConverter());
-        googleBookSender.sendBooksTo(serverIpAddress, "Google");
+        App app = new App();
+        Set<Scrapper> scrappers = new HashSet<>();
+        scrappers.add(new BonitoScrapper());
+        scrappers.add(new GandalfScrapper());
+        scrappers.add(new MatrasScrapper());
+        scrappers.add(new CzytamScrapper());
+        scrappers.add(new GoogleScrapper());
+        app.runApp(scrappers);
     }
 
-    private static Set<Book> runGoogleScrapper() {
-        GoogleScrapper googleScrapper = new GoogleScrapper();
-        googleScrapper.runScrapping();
-
-        while (!googleScrapper.areAllCallbacksDone()) {
-            sleepFor(6000, "");
+    void runApp(Set<Scrapper> scrappers) {
+        for (Scrapper scrapper : scrappers) {
+            Set<BookModel> books = scrapper.scrape();
+            sendScrappedBooks(books, scrapper.getName());
         }
```

Comment:
you can use java 8 feature scrappers.forEach( scrapper -> {Set<BookModel> books = scrapper.scrape(); 
sendScrappedBooks(books, scrapper.getName()})

Reply:
:+1: